packages/components/src/components/vertical-tab/vertical-tab-content-container.tsx

```diff
@@ -19,29 +40,29 @@ const SideNotes = ({ class_name, side_notes }) => {
     );
 };
 
-const ContentWrapper = ({ children, has_side_note }) => {
+const ContentWrapper = ({ children, has_side_note }: React.PropsWithChildren<TContentWrapper>) => {
     if (has_side_note) {
         return <div className='dc-vertical-tab__content-inner'>{children}</div>;
     }
-    return children;
+    return children as JSX.Element;
 };
 
-const Content = ({ is_routed, items, selected }) => {
+const Content = ({ is_routed, items, selected }: TContent) => {
     const selected_item = items.find(item => item.label === selected.label);
     const previous_selected_item = usePrevious(selected_item);
-    const TabContent = selected_item.value;
-    const [side_notes, setSideNotes] = React.useState(null);
+    const TabContent = selected_item?.value as React.ElementType;
+    const [side_notes, setSideNotes] = React.useState<React.ReactNode[] | null>(null);
 
-    const notes_array = [];
+    const notes_array: React.ReactNode[] = [];
 
-    const addToNotesQueue = React.useCallback(notes => {
+    const addToNotesQueue = React.useCallback((notes: React.ReactNode) => {
         notes_array.unshift(notes);
-        setSideNotes(notes);
+        setSideNotes(notes_array);
     }, []);
 
     React.useEffect(() => {
         if (selected_item?.label !== previous_selected_item?.label) {
-            setSideNotes(notes_array[0] ?? null);
+            setSideNotes(notes_array[0] ? notes_array : null);
```

Comment:
@yauheni-deriv why are we changing logics here? the scope of the card is TS migration

Reply:
setSideNotes is supposed to store the array (side_notes), but here we are setting item.  I suppose it was a typo here.
but I'll double check this.
thanks

Reply:
it seems there were some confusions for me. as I understand  now, notes_array is array of arrays, and we are setting only last notes for showing them.
will change it, thanks for your attention